documentation/library_guide/kernel_templates/esimd/radix_sort.rst

```diff
@@ -215,6 +252,132 @@ Usage Examples
    1 2 3 3 3 5
    s o r t e d
 
+``radix_sort`` out of place Example
+----------------------
+
+.. code:: cpp
```

Comment:
The examples section became disproportionally large compared to the rest documentation. But still, I do not treat it too large to block the PR, so the suggestions below are optional or/and may be done separately.

There are several things to consider:

- Make the examples collapsible. https://github.com/executablebooks/sphinx-togglebutton may be useful.
- Move the example to the bottom of the page.
- Move the example to the separate files and provide links.

Reply:
@dcbenito, is there any problem with using the collapsible examples?  I think I like that idea the best.

Reply:
I am also considering splitting that page in two parts: `radix_sort` and `radix_sort_by_key`. That will solve the issue. 

Let me try it. I will make a branch from this one: it should not take too much time. Then, we can decide if it worth continuing or we should do something different.

Reply:
I'm experimenting with the toggle stuff, and setting up my environment to be able to do the build locally rather than relying on the artifacts from the runners.
We can compare the two options.

Reply:
After struggling with getting sphinx togglebutton to render properly with a code block for a while, I don't think it is worth the trouble.  I also tried a custom css solution I found but that also has issues in combination with a code block.  

I'm sure there is a way to do it, but I think splitting the pages is the better option.  

Do you want to work to merge this PR first and then split? close this in favor of your combined PR, or target your PR at this branch?

Reply:
Let's merge this PR first: this PR is valuable as it is, we will keep the changes isolated.

Reply:
Meanwhile, I am working on splitting the page in #1586.

Reply:
I was worried that togglebutton wouldn't work. I think splitting the page makes sense.